### PR TITLE
2.7.3

### PR DESCRIPTION
**Pull request by uc-nfdz**
## Improvements

* General bug fixes.
* Huge performance improvement.

### CODE REVIEW

**Comment by uc-nfdz** [comment]
@mohammadshrahi - I've included your fix in this release, please check if this works properly. Thanks!

File: ios/Classes/API/UsercentricsBannerProxy.swift [hunk 1]
```diff
@@ -45,7 +45,7 @@ struct UsercentricsBannerProxy: UsercentricsBannerProxyProtocol {
         } else {
             window = UIApplication.shared.windows.first { $0.isKeyWindow }
         }
-        return window?.rootViewController as? FlutterViewController
+        return window?.rootViewController ?? window?.rootViewController?.presentedViewController
```
Comment (uc-nfdz) [inline ios/Classes/API/UsercentricsBannerProxy.swift:48]: @mohammadshrahi - Does this new approach cover your use case?

**Review by uc-leo — APPROVED** [review]

**Review by uc-leo — APPROVED** [review]

File: ios/Classes/API/UsercentricsBannerProxy.swift (hunk 1 above)
Comment (uc-nfdz) [reply ios/Classes/API/UsercentricsBannerProxy.swift:48]: I see, I understood the [presentedViewController](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621407-presentedviewcontroller) incorrectly. So what are you proposing? Should we navigate down to get the last `presentedViewController` by itereting until we find a nil?

Comment (mohammadshrahi) [reply ios/Classes/API/UsercentricsBannerProxy.swift:48]: Replacing         `return window?.rootViewController ?? window?.rootViewController?.presentedViewController`
with `window?.rootViewController?.presentedViewController`

Would solve the issue or?

**Pull request merged**